core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -161,13 +161,13 @@ private SqlQueryExecution(
             CostCalculator costCalculator,
             DynamicFilterService dynamicFilterService,
             WarningCollector warningCollector,
-            TableExecuteContextManager tableExecuteContextManager)
+            TableExecuteContextManager tableExecuteContextManager,
+            TypeAnalyzer typeAnalyzer)
     {
         try (SetThreadName ignored = new SetThreadName("Query-%s", stateMachine.getQueryId())) {
             this.slug = requireNonNull(slug, "slug is null");
             this.metadata = requireNonNull(metadata, "metadata is null");
             this.typeOperators = requireNonNull(typeOperators, "typeOperators is null");
-            this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
```

Comment:
If the `sqlParser` parameter is unused now, remove it

Reply:
It is used, in line 193.  it is passed to `analyze` method